src/components/Pagination/Pagination.tsx

```diff
@@ -1 +1,98 @@
-export const Pagination = () => {};
+import cn from 'classnames';
+import React, {
+  Dispatch, FC, SetStateAction,
+} from 'react';
+
+type Props = {
+  total: number,
+  perPage: number,
+  currentPage: number,
+  onPageChange: Dispatch<SetStateAction<number>>,
+};
+
+export const Pagination: FC<Props> = (
+  {
+    total,
+    perPage,
+    currentPage,
+    onPageChange: setCurrentPage,
+  },
+) => {
+  const totalPages = Math.ceil(total / perPage);
+  const pages = Array.from({ length: totalPages }, (_, i) => i + 1);
+  const isFirstPage = currentPage === pages[0];
+  const isLastPage = currentPage === pages.length;
+
+  const handleClickPrevNext = (
+    event: React.MouseEvent<HTMLAnchorElement, MouseEvent>,
+  ) => {
+    if (event
+      .currentTarget.href.includes('#next')
+    && !isLastPage) {
+      setCurrentPage(
+        (state) => state + 1,
+      );
+    } else if (event.currentTarget.href.includes('#prev')
+    && !isFirstPage) {
+      setCurrentPage(
+        (state) => state - 1,
+      );
+    }
```

Comment:
you can remove 'else', just make it `if`. Will look clearer

Reply:
✅